File: files/2026-05-24/CSEC 24052026.R
library(tidyverse)
library(readxl)

path <- "2026-05-24/Challenge 122.xlsx"
input <- read_excel(path, range = "B3:B11")
test <- read_excel(path, range = "D3:D11")

get_first_non_duplicate <- function(x) {
  chars <- strsplit(x, "", fixed = TRUE)[[1]]
  chars[which(!duplicated(chars) & !duplicated(chars, fromLast = TRUE))[1]]
}
result = input %>%
  mutate(first_non_duplicate = map_chr(Text, get_first_non_duplicate))

all.equal(result$first_non_duplicate, test$`1st Unique`)
# [1] TRUE

File: files/2026-05-24/CSEC 24052026.py
import pandas as pd
import numpy as np

path = "2026-05-24/Challenge 122.xlsx"
input = pd.read_excel(path, usecols="B", nrows=9, skiprows=2)
test = pd.read_excel(path, usecols="D", nrows=9, skiprows=2)

def first_nonduplicated_char(s):
    for c in s:
        if s.count(c) == 1:
            return c
    return np.nan

input["First Nonduplicated Character"] = input["Text"].apply(first_nonduplicated_char)
print(input["First Nonduplicated Character"].equals(test['1st Unique']))
# True